[FLINK-17125][python] Add a Usage Notes Page to Answer Common Questions Encountered by PyFlink Users (#11878)
apache/flink: docs/dev/table/python/common_questions.md

---
title: "Common Questions"
nav-parent_id: python_tableapi
nav-pos: 140
---
<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

> This page describes the solutions to some frequently encountered problems for PyFlink users.

Reviewer: ```suggestion
This page describes the solutions to some come questions for PyFlink users.
```

* This will be replaced by the TOC
{:toc}
## Preparing Python Virtual Environment

> You can download a [convenience script](https://github.com/apache/flink/blob/master/flink-python/dev/setup-pyflink-virtual-env.sh) to prepare a Python virtual env zip which can be used on Mac OS and most Linux distributions. 

Reviewer: ```suggestion
You can download the [convenience script](https://github.com/apache/flink/blob/master/flink-python/dev/setup-pyflink-virtual-env.sh) to prepare a Python virtual env zip which can be used on Mac OS and most Linux distributions. 
```

under the License.
-->

This page describes the solutions to some common questions for PyFlink users.

* This will be replaced by the TOC
{:toc}
## Preparing Python Virtual Environment
You can download a [convenience script](setup-pyflink-virtual-env.sh) to prepare a Python virtual env zip which can be used on Mac OS and most Linux distributions.

You can specify the version parameter to generate a Python virtual environment required for the corresponding PyFlink version. e.g.

{% highlight shell %}
# you will get a Python virtual environment required by PyFlink version 1.10

> setup-pyflink-virtual-env.sh 1.10

Reviewer 2: This way the version is always up to date. 

```suggestion
setup-pyflink-virtual-env.sh {{ site.version }}
```

Author: Or I support a specified version example and a no version example

Author: {% highlight shell %}
# you will get a Python virtual environment required by PyFlink version 1.10
setup-pyflink-virtual-env.sh 1.10

# you will get a Python virtual environment required by PyFlink with the latest version
setup-pyflink-virtual-env.sh
{% endhighlight %}

Reviewer 2: How about this: 

{% if site.is_stable %}
$ setup-pyflink-virtual-env.sh {{ site.version }}
{% else %}
$ setup-pyflink-virtual-env.sh
{% endif %}

Author: It looks better.I will update soon.Thanks.
